homeassistant/components/flick_electric/diagnostics.py

```diff
@@ -0,0 +1,25 @@
+"""Diagnostics for the Flick Electric integration."""
+
+from typing import Any
+
+from homeassistant.const import CONF_CLIENT_SECRET, CONF_PASSWORD
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.redact import async_redact_data
+
+from .coordinator import FlickConfigEntry
+
+TO_REDACT = [
+    CONF_PASSWORD,
+    CONF_CLIENT_SECRET,
+]
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, entry: FlickConfigEntry
+) -> dict[str, Any]:
+    """Return diagnostics for a config entry."""
+
+    return {
+        "entry_data": async_redact_data(entry.data, TO_REDACT),
+        "data": entry.runtime_data.data,
```

Comment:
Should the diagnostics return a "pure" object, or is it fine to return class types here? Spotted it while looking at the snapshot data.

I can do `data: entry.runtime_data.data.raw` instead - but that's basically what `FlickPrice.__repr__` already does.

Reply:
I think it is not a big deal. Compared to:
https://github.com/home-assistant/core/blob/15223b36794a6e2399e6434b339e2c9f57a424f3/tests/components/incomfort/snapshots/test_diagnostics.ambr
It seems the snapshot tries to make representation including the types exposed.

Reply:
Is it a dataclass?

Wrap it in a `as_dict`